File: src/convex/Convex.java
package convex;

import convex.geometry.Point;

class Convex {
    private convex.figure.Figure figure;
    private convex.geometry.Point task;
    Convex() throws Exception {
        System.out.println("Enter task point");
        task = new Point();
        System.out.println("Enter convex points");
        figure = new convex.figure.Void();
    }
    void add(Point point) {
        figure = figure.add(point, task);
    }
    double area() {
        return figure.area();
    }
    double perimeter() {
        return figure.perimeter();
    }
    double result() {
        return figure.result();
    }
}

File: Convex/Convex.cs
using Convex.Figure;

namespace Convex
{
    internal class Convex
    {
        private IFigure figure;

        public Convex()
        {
            figure = new Void();
        }

        public void Add(Geometry.Point point)
        {
            figure = figure.Add(point);
        }

        public double Area
        {
            get => figure.Area;
        }

        public double Perimeter
        {
            get => figure.Perimeter;
        }
    }
}
